### PR TITLE
Feat/contribution

### PR DESCRIPTION
**Pull request by mysticknits**

### CODE REVIEW

File: contracts/Token.sol [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 
-contract Token is ERC20, ERC20Detailed {
+contract Token is Ownable, ERC20 {
   uint40 public startTime;
   uint40 public endTime;
 
-  mapping(address => Token) public tokens;
-  constructor(uint40 _startTime, uint40 _endTime, uint256 initialSupply) ERC20Detailed("Gold", "GLD", 18) public {
+  constructor(uint40 _startTime, uint40 _endTime, uint256 initialSupply) public {
```
Comment (kseniya292) [inline contracts/Token.sol:12]: @laurenamy why did you get rid of the ERC20Detailed contract btw?

File: test/Token.test.js [hunk 2]
```diff
@@ -1,34 +1,64 @@
...
+
+    contribution = await Contribution.new(tokenAddress);
+    contributionAddress = contribution.address;
+    contributionInstance = await Contribution.at(contributionAddress);
+
+    // owner gives approval to contributionContract to transfer on their behalf
+    await token.approve(contributionAddress, tokenSupply);
   });
 
   describe('transfer', function () {
```
Comment (kseniya292) [inline test/Token.test.js]: update to `transferFrom`

File: test/Token.test.js [hunk 3]
```diff
@@ -1,34 +1,64 @@
...
     // Saturday, August 1, 2020 9:00:00 AM
     tokenAddress = token.address;
     tokenInstance = await Token.at(tokenAddress);
+
+    contribution = await Contribution.new(tokenAddress);
+    contributionAddress = contribution.address;
+    contributionInstance = await Contribution.at(contributionAddress);
+
+    // owner gives approval to contributionContract to transfer on their behalf
+    await token.approve(contributionAddress, tokenSupply);
```
Comment (kseniya292) [inline test/Token.test.js:30]: see how we are using `token` here rather then `tokenInstance`? I'm not sure why we even use `tokenInstance = await Token.at(tokenAddress);` Seems like we can use `token` and `contribution`. Want to update everywhere to be consistent?

File: test/Token.test.js [hunk 4]
```diff
@@ -1,34 +1,64 @@
...
   });
 
   describe('transfer', function () {
-    it('emits a Transfer event on successful Transfers', async function () {
-      const { logs } = await token.transfer(recipient, amount, { from: owner });
-      await expectEvent.inLogs(logs, 'Transfer', {to: recipient, value: amount });
+    it('emits a Sent event on successful Transfers', async function () {
```
Comment (kseniya292) [inline test/Token.test.js]: you are already testing for this in Contribution.test.js so you can remove this.

**Review by kseniya292 — COMMENTED** [review]
Looks great! A couple more tests you could add in `Token.test.js`
it should set startTime
it should set endTime
it should update startTime (when setStartDate is called)
it should updated endDate (when setEndDate is called)

File: contracts/Token.sol (hunk 1 above)
Comment (mysticknits) [reply contracts/Token.sol:12]: It wasn't really serving a purpose and I didn't see a reason to include it in the future issues, but I can add it back in if necessary!

**Pull request merged**